metadata-ingestion/tests/unit/test_vertexai_source.py

```diff
@@ -0,0 +1,446 @@
+    mock_model_1.version_id = "1"
+    mock_model_1.display_name = "mock_prediction_model_1_display_name"
+    mock_model_1.resource_name = (
+        "projects/872197881936/locations/us-west2/models/3583871344875405312"
+    )
+    return mock_model_1
+
+
+def gen_mock_models() -> List[Model]:
+    mock_model_1 = MagicMock(spec=Model)
+    mock_model_1.name = "mock_prediction_model_1"
+    mock_model_1.create_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_model_1.update_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_model_1.version_id = "1"
+    mock_model_1.display_name = "mock_prediction_model_1_display_name"
+    mock_model_1.description = "mock_prediction_model_1_description"
+
+    mock_model_2 = MagicMock(spec=Model)
+    mock_model_2.name = "mock_prediction_model_2"
+
+    mock_model_2.create_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_model_2.update_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_model_2.version_id = "1"
+    mock_model_2.display_name = "mock_prediction_model_2_display_name"
+    mock_model_2.description = "mock_prediction_model_1_description"
+
+    return [mock_model_1, mock_model_2]
+
+
+def gen_mock_training_job() -> VertexAiResourceNoun:
+    mock_training_job = MagicMock(spec=VertexAiResourceNoun)
+    mock_training_job.name = "mock_training_job"
+    mock_training_job.create_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_training_job.update_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_training_job.display_name = "mock_training_job_display_name"
+    mock_training_job.description = "mock_training_job_description"
+    return mock_training_job
+
+
+def gen_mock_dataset() -> VertexAiResourceNoun:
+    mock_dataset = MagicMock(spec=VertexAiResourceNoun)
+    mock_dataset.name = "mock_dataset"
+    mock_dataset.create_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_dataset.update_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_dataset.display_name = "mock_dataset_display_name"
+    mock_dataset.description = "mock_dataset_description"
+    return mock_dataset
+
+
+def gen_mock_training_automl_job() -> AutoMLTabularTrainingJob:
+    mock_automl_job = MagicMock(spec=AutoMLTabularTrainingJob)
+    mock_automl_job.name = "mock_auto_automl_tabular_job"
+    mock_automl_job.create_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_automl_job.update_time = timestamp_pb2.Timestamp().GetCurrentTime()
+    mock_automl_job.display_name = "mock_auto_automl_tabular_job_display_name"
+    mock_automl_job.description = "mock_auto_automl_tabular_job_display_name"
+    return mock_automl_job
+
+
+def gen_mock_endpoint() -> Endpoint:
+    mock_endpoint = MagicMock(spec=Endpoint)
+    mock_endpoint.description = "test endpoint"
+    mock_endpoint.create_time = datetime.now()
+    mock_endpoint.display_name = "test endpoint display name"
+    return mock_endpoint
```

Comment:
is this stuff copy-pasted from the other test file? if so, we should probably pull it into a vertex_ai_mocks.py that both import